src/drivers/intel/baytrail/ssp.c

```diff
@@ -525,10 +547,13 @@ static void ssp_start(struct dai *dai, int direction)
 	dai_info(dai, "ssp_start()");
 
 	/* enable DMA */
-	if (direction == DAI_DIR_PLAYBACK)
+	if (direction == DAI_DIR_PLAYBACK) {
 		ssp_update_bits(dai, SSCR1, SSCR1_TSRE, SSCR1_TSRE);
-	else
+		ssp_update_bits(dai, SSTSA, SSTSA_TXEN, SSTSA_TXEN);
```

Comment:
iiuc, we don't have TXEN/RXEN bit on BYT SSP, which means we can't stop TX/RX separately there, if you do a while() check to the TFL/RFL in the FIFO flush, you will find this sad result.

Reply:
sorry @keyonjie not understanding your feedback. 
There's clearly separate registers for SSTSA_TXEN and SSRSA_RXEN.

What are you referring to?

Reply:
@plbossart I guess you are referring to the wrong register map, for BYT, we don't have SSTSA_TXEN and SSRSA_TXEN bits, though we have TTSA abd RTSA there.

Reply:
@keyonjie I checked the hardware documentation.

Reply:
Actually now I am confused, I only see RTSA and TTSA but I don't see where we set them. Humm.